### PR TITLE
NXP-24997: Add Functional Tests to delegate and reassign task options

### PR DESCRIPTION
**Pull request by nmpcunha**

### CODE REVIEW

File: nuxeo-web-ui-ftest/test/features/step_definitions/document.js [hunk 1]
```diff
@@ -142,7 +182,7 @@ module.exports = function () {
         }
         break;
       default:
-        // do nothing
+      // do nothing
```
Comment (Gabez0r) [inline nuxeo-web-ui-ftest/test/features/step_definitions/document.js]: nit: weird indentation

File: nuxeo-web-ui-ftest/test/features/step_definitions/document.js [hunk 2]
```diff
@@ -28,32 +28,81 @@ module.exports = function () {
...
+              let value;
+              switch (row[2]) {
+                case 'list':
+                  value = row[1].split(',');
+                  break;
+                case 'text':
+                  value = row[1];
+                  break;
+                default:
+                  fail(row[2], '', 'An unknown type was passed as argument', '');
```
Comment (Gabez0r) [inline nuxeo-web-ui-ftest/test/features/step_definitions/document.js]: We should not need to import fail and stick with expect/should if we use [oneOf](http://www.chaijs.com/api/bdd/#method_oneof) assertion. We could do something like before the switch (we may even go without the switch):
```
expect(row[2]).to.be.oneOf(['list', 'text'], 'An unknown type was passed as argument');
```

File: nuxeo-web-ui-ftest/test/features/step_definitions/tasks.js [hunk 3]
```diff
@@ -1,9 +1,75 @@
...
+  this.When(/^I (\w+) the task for following actors:$/, (option, table) => {
+    switch (option) {
+      case 'delegate':
+        this.ui.browser.documentTaskView.delegateOption.click();
+        break;
+      case 'reassign':
+        this.ui.browser.documentTaskView.reassignOption.click();
+        break;
+      default:
+        fail(option, '', 'An unknown option was passed as argument', '');
```
Comment (Gabez0r) [inline nuxeo-web-ui-ftest/test/features/step_definitions/tasks.js]: Same as above, use [oneOf](http://www.chaijs.com/api/bdd/#method_oneof) instead. Consider using `if` instead.

File: nuxeo-web-ui-ftest/test/features/step_definitions/tasks.js [hunk 4]
```diff
@@ -1,9 +1,75 @@
...
+  this.Then(/^I can see the (\w+) option available$/, (option) => {
+    switch (option) {
+      case 'delegate':
+        this.ui.browser.documentTaskView.delegateOption.isVisible().should.be.true;
+        break;
+      case 'reassign':
+        this.ui.browser.documentTaskView.reassignOption.isVisible().should.be.true;
+        break;
+      default:
+        fail(option, '', 'An unknown option was passed as argument', '');
```
Comment (Gabez0r) [inline nuxeo-web-ui-ftest/test/features/step_definitions/tasks.js]: Same as above, use [oneOf](http://www.chaijs.com/api/bdd/#method_oneof) instead. Consider using `if` instead.

File: nuxeo-web-ui-ftest/test/features/step_definitions/tasks.js [hunk 5]
```diff
@@ -1,9 +1,75 @@
...
+  this.Then(/^I can't see the (\w+) option available$/, (option) => {
+    switch (option) {
+      case 'delegate':
+        this.ui.browser.documentTaskView.delegateOption.isVisible().should.be.false;
+        break;
+      case 'reassign':
+        this.ui.browser.documentTaskView.reassignOption.isVisible().should.be.false;
+        break;
+      default:
+        fail(option, '', 'An unknown option was passed as argument', '');
```
Comment (Gabez0r) [inline nuxeo-web-ui-ftest/test/features/step_definitions/tasks.js]: Same as above, use [oneOf](http://www.chaijs.com/api/bdd/#method_oneof) instead. Consider using `if` instead.

File: nuxeo-web-ui-ftest/test/features/step_definitions/tasks.js [hunk 6]
```diff
@@ -1,9 +1,75 @@
...
+    let actorsElement;
+    switch (option) {
+      case 'delegated':
+        actorsElement = documentTaskView.delegatedActors;
+        break;
+      case 'assigned':
+        actorsElement = documentTaskView.assignedActors;
+        break;
+      default:
+        fail(option, '', 'An unknown option was passed as argument', '');
```
Comment (Gabez0r) [inline nuxeo-web-ui-ftest/test/features/step_definitions/tasks.js]: Same as above, use [oneOf](http://www.chaijs.com/api/bdd/#method_oneof) instead. Consider using `if` instead.

**Comment by Gabez0r** [comment]
👍

**Comment by guirenard** [comment]
:+1:

**Comment by nmpcunha** [comment]
https://qa.nuxeo.org/jenkins/view/TestAndPush/job/web-ui-pipeline-ncunha/8/

**Pull request merged**